tests/main/views/test_drafts.py

```diff
@@ -50,12 +51,396 @@ def setup(self):
             **payload
         )
 
+    def teardown(self):
+        super().teardown()
+
     def service_count(self):
         return Service.query.count()
 
     def draft_service_count(self):
         return DraftService.query.count()
 
+    def create_draft_service(self):
+        res = self.client.post(
+            '/draft-services',
+            data=json.dumps(self.create_draft_json),
+            content_type='application/json')
+        assert res.status_code == 201
+        draft = json.loads(res.get_data())['services']
+
+        g7_complete = load_example_listing("G7-SCS").copy()
+        g7_complete.pop('id')
+        draft_update_json = {'services': g7_complete,
+                             'updated_by': 'joeblogs'}
+        res2 = self.client.post(
+            '/draft-services/{}'.format(draft['id']),
+            data=json.dumps(draft_update_json),
+            content_type='application/json')
+        assert res2.status_code == 200
+        draft = json.loads(res2.get_data())['services']
+
+        return draft
+
+    def complete_draft_service(self, draft_id):
+        return self.client.post(
+            '/draft-services/{}/complete'.format(draft_id),
+            data=json.dumps({'updated_by': 'joeblogs'}),
+            content_type='application/json')
+
+    def publish_draft_service(self, draft_id):
+        return self.client.post(
+            '/draft-services/{}/publish'.format(draft_id),
+            data=json.dumps({
+                'updated_by': 'joeblogs'
+            }),
+            content_type='application/json')
+
+    def publish_new_draft_service(self):
+        draft = self.create_draft_service()
+        res = self.complete_draft_service(draft['id'])
+        assert res.status_code == 200
+
+        res = self.publish_draft_service(draft['id'])
+        assert res.status_code == 200
+
+        return res
+
+
+class TestCopyDraftServiceFromExistingService(DraftsHelpersMixin):
+    def test_reject_copy_with_no_updated_by(self):
+        res = self.client.put('/draft-services/copy-from/0000000000')
+        assert res.status_code == 400
```

Comment:
I've just checked and you're right, it's actually failing because of an unexpected content type! These tests I just split out from the existing class (cos it was over 1000 lines) so I might need to look at if they're actually doing what they're supposed to.